File: DBGuardClient/src/app/interfaces/guard-dto.ts
import { DatabaseEngine } from "../enums/database-engines";
import { GuardOperator } from "../enums/guard-operator";
import { GuardState } from "../enums/guard-state";
import { NotificationType } from "../enums/notification-type";
import { DatabaseConnectionDTO } from "./database-connection-dto";
import { CreateGuardNotificationDTO } from "./notification-dto";
import { NotificationProviderDTO } from "./notification-provider-dto";

export interface CreateGuardDTO {
    id?: number,
    guardName?: string,
    guardDescription?: string,
    triggerQuery: string,
    countColumn: string,
    triggerOperator: GuardOperator
    triggerValue: number,
    databaseConnection: DatabaseConnectionDTO,
    notifyOnClear: boolean,
    notifyOnError: boolean,
    notifyOnTrigger: boolean,
    runPeriodInMinutes: number,
    notifications: CreateGuardNotificationDTO[]
}

export interface CreateGuardReferenceData {
    databaseConnections: DatabaseConnectionDTO[],
    notificationProviders: NotificationProviderDTO[]
}

export interface GuardDTO {
    id: number,
    guardName?: string,
    guardDescriptio?: string
}
export interface GuardView {
    id: number,
    guardName?: string,
    createDate: Date,
    lastRun: Date,
    createdByUserId: string,
    userName: string,
    countColumn: string,
    triggerOperator: GuardOperator,
    triggerValue: number,
    databaseConnectionId: number,
    endPoint: string,
    databaseEngine: DatabaseEngine,
    databaseName: string,
    guardState: GuardState,
    isActive: boolean,
    totalErrors: number,
    totalTriggers: number,
    runPeriodInMinutes: number
}

export interface GuardDetailDTO {
    id: number,
    guardName?: string,
    guardDescription?: string,
    createDate: Date,
    lastRun?: Date,
    lastEditedDate: Date,
    createdByUserId: string,
    userName: string,
    triggerQuery: string,
    countColumn: string,
    triggerOperator: GuardOperator,
    triggerValue: number,
    guardState: GuardState,
    isActive: boolean,
    notifyOnClear: boolean,
    notifyOnError: boolean,
    notifyOnTrigger: boolean,
    totalErrors: number,
    totalTriggers: number,
    runPeriodInMinutes: number
}

File: DBGuardAPI/Data/DTOs/GuardDTOs/GuardDTO.cs
using DBGuardAPI.Data.Enums;
using Sieve.Attributes;

namespace DBGuardAPI.Data.DTOs.GuardDTOs
{
    public class GuardDTO
    {
        [Sieve(CanSort = true, CanFilter = true)]
        public int Id { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public string? GuardName { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DateTimeOffset CreateDate { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DateTimeOffset? LastRun { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DateTimeOffset LastEditedDate { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string CreatedByUserId { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string UserName { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string TriggerQuery { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string CountColumn { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public GuardOperator TriggerOperator { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int TriggerValue { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public GuardState GuardState { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public bool IsActive { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public bool NotifyOnClear { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public bool NotifyOnError { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public bool NotifyOnTrigger { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int TotalErrors { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int TotalTriggers { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public double RunPeriodInMinutes { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int DatabaseConnectionId { get; set; }
    }
}
